Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -289,5 +292,18 @@ public void RegisterForDisposal(IDisposable disposable)
         public void Dispose() => _disposables.Dispose();
 
         string IKernel.Name => Name;
+
+        public void SetProperty<T>(T property) where T : class
+        {
+            if (!_propertyBag.TryAdd(typeof(T), property))
```

Comment:
If you use `Add` you can just let the default exception be thrown.

Reply:
I don’t think there is Add on concurrent dictionary